### DISCUSSION TITLE
Warning: Circular dependency detected. `codeReordering` pragma may not be able to reorder some nodes properly [User]

### DISCUSSION THREAD

**Discussion by jerrygzy** [HapticX/happyx#360] [Q&A]
got this after build

```
/root/.nimble/pkgs2/happyx-4.6.2-c009f8e146fe778822f58d0ac65941816ee8305f/happyx/spa/renderer.nim(22, 1) Warning: Circular dependency detected. `codeReordering` pragma may not be able to reorder some nodes properly [User]
```

and I am not sure why, after upgraded to 4.6.2, there is a duplicated function call in my log.

```
[2024-10-09T11:33:02+08:00] private POST request /search
query result count: 100
query result count: 100
```

switch back to 4.5.2, does not have above problem
```
[2024-10-09T11:47:42+08:00] private POST request /search
query result count: 100
```

**Comment by Ethosa** [comment]
please provide more information, ex. code

**Reply by jerrygzy** [reply]
I think it maybe better to send you the project code, it will be easier for debugging. but since the code is part of company product, can I remove sensitive information and send to your email?

**Reply by Ethosa** [reply]
yes you can send it to hapticx.company@gmail.com
